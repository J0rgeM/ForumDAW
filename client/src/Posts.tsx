import React, { useEffect, useState } from "react";
// Library imports.
import axios, { AxiosResponse } from "axios";
import Navbar from "./components/Navbar";
import Footer from "./components/Footer";

import { IPost } from "../../server/src/posts";
import { Link } from "react-router-dom";

function Posts() {
    const [posts, setPosts] = useState<IPost[]>([]);

    const getPost = async () => {
        try {
            const response: AxiosResponse = await axios.get<IPost[]>(
                "http://localhost:8080/"
            );
            response.data.sort((n1: IPost, n2: IPost) => n1.nmb - n2.nmb);
            setPosts(response.data);
            console.log(response.data);
        } catch (error) {
            console.error(error);
        }
    };
    useEffect(() => {
        getPost();
    }, []);

    const deletePost = async (number) => {
        try {
            const response: AxiosResponse = await axios.delete(
                "http://localhost:8080/".concat(number)
            );
            console.log(response);
            window.location.reload();
        } catch (error) {
            console.error(error);
        }
    };

    return (
        <div>
            <Navbar />
            <div className="container text-center">
                <div className="margin-0 margin-top-0 btn-group" role="group">
                    <Link to="/addPost">
                        <button className="m-2 p-2 btn btn-dark">
                            Add Post
                        </button>
                    </Link>
                </div>
            </div>
            {posts.map((post) => (
                <div
                    key={post.nmb}
                    className="container shadow p-3 mb-5 bg-white rounded"
                >
                    <div className="col">
                        <div className="card">
                            <div className="row g-0">
                                <div className="col-md-4">
                                    <img
                                        src="https://media.giphy.com/media/sQuHLqjWwRXGvrjkg0/giphy.gif"
                                        width="100%"
                                        height={250}
                                        alt="meme"
                                    />
                                </div>
                                <div className="col-md-8">
                                    <div className="card-body">
                                        <p className="card-text">
                                            number: {post.nmb}
                                        </p>
                                        <p className="card-text">
                                            Author: {post.author}
                                        </p>
                                        <p className="card-text">{post.body}</p>
                                        <div className="float-end">
                                            <Link to={"/updatePost/" + post.nmb}>
                                                <button className="m-2 p-2 btn btn-dark">
                                                    Update Post
                                                </button>
                                            </Link>
                                            <button
                                                type="submit"
                                                onClick={() =>
                                                    deletePost(post.nmb)
                                                }
                                                className="m-2 p-2 btn btn-outline-dark"
                                            >
                                                Delete Post
                                            </button>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            ))}
            <Footer />
        </div>
    );
}

export default Posts;
