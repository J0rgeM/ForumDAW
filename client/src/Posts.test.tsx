import {describe, expect, it} from '@jest/globals';
import {renderWithRouter} from "./lib";
import Posts from "./Posts";

describe('Posts', () => {
    it('match snapshot', () => {
        const { container} = renderWithRouter(<Posts/>)
        expect(container).toMatchSnapshot()
    })
})