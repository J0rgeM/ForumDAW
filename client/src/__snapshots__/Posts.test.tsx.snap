// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts match snapshot 1`] = `
<div>
  <div>
    <nav
      class="navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
           BlogGIF
        </a>
        <button
          aria-controls="navbarNavDropdown"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-bs-target="#navbarNavDropdown"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse justify-content-end"
          id="navbarNavDropdown"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item"
            >
              <a
                aria-current="page"
                class="nav-link active rounded-3"
                href="/addPost"
              >
                Add Post
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                aria-current="page"
                class="nav-link active rounded-3"
                href="/"
              >
                List Posts
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div
      class="container text-center"
    >
      <div
        class="margin-0 margin-top-0 btn-group"
        role="group"
      >
        <a
          href="/addPost"
        >
          <button
            class="m-2 p-2 btn btn-dark"
          >
            Add Post
          </button>
        </a>
      </div>
    </div>
    <div
      class="footer text-center fixed-bottom text-white bg-dark"
    >
      <div
        class="container-fluid"
      >
        <div
          class="row"
        >
          <p
            class="text-center"
          >
            Â© 2023 DAW
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
